### PR TITLE
Add test for partial joint trajectory client

### PR DESCRIPTION
**Pull request by chatblanc-ciel**
creating test function for partial joint trajectory client

### CODE REVIEW

File: arci/src/clients/partial_joint_trajectory_client.rs [hunk 1]
```diff
@@ -103,3 +103,249 @@ where
...
+            ),
+            TrajectoryPoint::new(
+                vec![3.4_f64, 5.8_f64, 0.1_f64, 2.5_f64],
+                std::time::Duration::from_secs(2),
+            ),
+        ];
+        let correct = vec![3.4_f64, 5.8_f64, 0.1_f64, 2.5_f64];
+
+        let partial = PartialJointTrajectoryClient::new(joint_names.clone(), client.clone());
+        let result = partial.send_joint_trajectory(trajectories);
```
Comment (chatblanc-ciel) [inline arci/src/clients/partial_joint_trajectory_client.rs:340]: if trajectories length is different from client name length,
I think ``fn send_joint_trajectory`` needs assert macro.

Now, part of trajectories are copied and others are zero.
length of the part of trajectories is same as client name length.

**Review by chatblanc-ciel — COMMENTED** [review]
purpose

**Comment by a-miyadera** [comment]
In actual situation, the length of joint(_name)s of ParitialJointTrajectoryClinet is shorter than that of shared_client (full JointTrajectoryClient).
So, we may need tests in that condition.

File: arci/src/clients/partial_joint_trajectory_client.rs [hunk 2]
```diff
@@ -103,3 +103,410 @@ where
...
+        assert_eq!(
+            format!("{:?}", client),
+            format!("{:?}", partial.shared_client)
+        );
+        assert_eq!(
+            format!("{:?}", partial.full_joint_names),
+            "[\"part1\", \"high\", \"part2\"]"
+        );
+    }
+
```
Comment (a-miyadera) [inline arci/src/clients/partial_joint_trajectory_client.rs:202]: What's the difference between `fn test_fn_copy_joint_position_for_from_joint()` and `fn test_fn_copy_joint_position_for_to_joint()`?

**Review by a-miyadera — APPROVED** [review]

**Pull request merged**